audio-analysis/lib/utils/videoHelper.py

```diff
@@ -79,23 +90,50 @@ def getTextAnalysis(self):
         with sr.AudioFile(AUDIO_FILE) as source:
             audio = r.record(source)  # read the entire audio file
             decoder = r.recognize_sphinx(audio, show_all=True)
+            decoded_words = decoder.seg()
+            video_title = self.info_dict.get('title', None)
+            overlap = 2
+            slide_inc = self.windowSize - overlap
+            print(slide_inc)
+            if overlap > self.windowSize:
+                raise Exception('ERROR - overlap must not exceed windowSize.')
+            num_windows = self.getDuration() // slide_inc
             print(decoder.seg())
 
-            # Prints out all the words and their start and end timestamps.
-            for seg in decoder.seg():
-                if (self.isTest == True):
-                    print(seg.word, seg.start_frame, seg.end_frame)
-                video_title = self.info_dict.get('title', None)
-                word_dict = {
-                                 'word': seg.word,
-                                 'start_time_s': math.trunc(seg.start_frame/100),
-                                 'end_time_s': math.trunc(seg.end_frame/100),
-                                 'subjectivity': TextBlob(seg.word).sentiment.subjectivity,
-                                 'polarity': TextBlob(seg.word).sentiment.polarity,
+            for i in range(0, num_windows):
```

Comment:
I think this would make too many windows?? ex: `duration = 10`, then `num_windows = 10 `, `start_times = 0, 2, 4, 6, 8, 10, 12 ... 20`

Reply:
should add a `if end_time > duration: endtime = duration` then after you collect the data you can do `if end_time >= duration: break`

Reply:
Mmmm - well no. It would not make too many windows - although there was a logic fencepost problem with the final widow.

Reply:
Ohhhh ok - now I get what you meant `if end_time > duration: endtime = duration`